package.json

```diff
@@ -104,17 +104,18 @@
     "check-licenses": "wp-scripts check-licenses --production",
     "deploy": "grunt deploy",
     "dev": "wp-scripts start",
+    "diff-develop": "git diff develop --diff-filter=d --name-only",
     "env:start": "./bin/local-env/start.sh",
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff-develop | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "npm run diff-develop | grep -E '\\.js$' | xargs wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "vendor/bin/phpcs",
+    "lint:php": "npm run diff-develop | grep -E '\\.php$' | xargs vendor/bin/phpcs",
```

Comment:
This isn't working properly when there are no changed PHP files. It is ending up checking _all_ files. Perhaps it is because the `phpcs.xml` has `<file>.</file>`?

Reply:
Hm, it looks like that might have been because `npm run diff-develop` also takes account of the changes in the `develop` branch, and this branch didn't have the latest from `develop`.

Before the [commit below](https://github.com/ampproject/amp-wp/pull/4441/commits/7d35852b1a1efa74b1dc71acd1162bc11e0d7edd), here's the result of `npm run diff-develop`

```sh
bin/tag-built.sh
composer.json
composer.lock
contributing/engineering.md
includes/sanitizers/class-amp-base-sanitizer.php
includes/validation/class-amp-validation-error-taxonomy.php
lib/common/src/DevMode.php
lib/common/src/Dom/Document.php
lib/common/tests/DevModeTest.php
lib/common/tests/Dom/DocumentTest.php
package-lock.json
package.json
tests/php/test-amp-dev-mode-sanitizer.php
tests/php/test-class-amp-base-sanitizer.php
tests/php/test-tag-and-attribute-sanitizer.php
tests/php/validation/test-class-amp-validation-error-taxonomy.php
tests/php/validation/test-class-amp-validation-manager.php
```

Reply:
Still, it's not good that it lints the files that were changed in `develop`, as those aren't relevant here.